commit 5f37ab207ac330ac1cc2b3136793e2bcd4fbb2b4
Author: Balló György <ballogyor@gmail.com>
Date:   Tue Oct 7 21:23:13 2014 +0200

    lxsession-db: Don't hang when MIME folders are not available.

diff --git a/lxsession-db/desktop-files-backend.vala b/lxsession-db/desktop-files-backend.vala
index d68644d..a9a52e8 100644
--- a/lxsession-db/desktop-files-backend.vala
+++ b/lxsession-db/desktop-files-backend.vala
@@ -315,21 +315,21 @@ namespace Lxsession
                         }
                     }
                 }
-
-                if (mode == "installed")
-                {
-                    finish_scanning_installed();
-                }
-                else if (mode == "available")
-                {
-                    finish_scanning_available();
-                }
-
             } 
             catch (Error err)
             {
                 stderr.printf ("Error: list_files failed: %s\n", err.message);
             }
+
+            if (mode == "installed")
+            {
+                finish_scanning_installed();
+            }
+            else if (mode == "available")
+            {
+                finish_scanning_available();
+            }
+
             message ("Finishing scanning\n");
         }
 
